src/test/operators/join_index_test.cpp

```diff
@@ -0,0 +1,96 @@
+#include <map>
+#include <memory>
+#include <numeric>
+#include <utility>
+#include <vector>
+
+#include "../base_test.hpp"
+#include "gtest/gtest.h"
+
+#include "operators/join_index.hpp"
+#include "operators/table_wrapper.hpp"
+#include "storage/dictionary_compression.hpp"
+#include "storage/index/adaptive_radix_tree/adaptive_radix_tree_index.hpp"
+#include "storage/index/group_key/composite_group_key_index.hpp"
+#include "storage/index/group_key/group_key_index.hpp"
+#include "storage/table.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+template <typename DerivedIndex>
+class OperatorsJoinIndexTest : public BaseTest {
+ protected:
+  void SetUp() override {
+    _index_type = get_index_type_of<DerivedIndex>();
+
+    auto left_table = std::make_shared<Table>(5);
+    left_table->add_column("a", DataType::Int);
+    left_table->add_column("b", DataType::Int);
+    for (int i = 0; i <= 24; i += 2) left_table->append({i, 100 + i});
+    DictionaryCompression::compress_table(*left_table);
+
+    auto right_table = std::make_shared<Table>(5);
+    right_table->add_column("a", DataType::Int);
+    right_table->add_column("b", DataType::Int);
+    for (int i = 0; i <= 24; i += 2) right_table->append({i, 100 + i});
+    DictionaryCompression::compress_table(*right_table);
+
+    _chunk_ids = std::vector<ChunkID>(left_table->chunk_count());
+    std::iota(_chunk_ids.begin(), _chunk_ids.end(), ChunkID{0u});
+
+    _column_ids = std::vector<ColumnID>{ColumnID{0u}};
+
+    for (const auto& chunk_id : _chunk_ids) {
+      auto& chunk = left_table->get_chunk(chunk_id);
+      chunk.template create_index<DerivedIndex>(_column_ids);
```

Comment:
We copied this from another point, but just now realized we don't need it because chunk is not a templated type here.
See https://stackoverflow.com/questions/3499101/when-do-we-need-a-template-construct for further information

Reply:
To me, it doesn't even look like valid c++ at all.

Reply:
Yes, that was our reaction at first as well...